### ISSUE TITLE
eval_stats causes mixmatch to freeze

### ISSUE THREAD

**Issue by DewaldHoman** [google-research/mixmatch#14]
When using mixmatch.py to train a dataset (e.g. cifar10 README example), the summary generated by `eval_stats` seems to freeze the program after a few epochs. The problem seems to be introduced at the start of evaluating the `valid` subset, during the loop:

`for subset in ('train_labeled', 'valid', 'test'):`
 &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;`images, labels = self.tmp.cache[subset]`
 &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;`predicted = []`
 &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;`for x in range(0, images.shape[0], batch):`
 &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;`p = self.session.run(`
 &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;`classify_op,`
 &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;`feed_dict={`
 &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;`self.ops.x: images[x:x + batch],`
 &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;`**(feed_extra or {})`
 &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;`})`
 &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;`predicted.append(p)`
 &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;`predicted = np.concatenate(predicted, axis=0)`
 &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;`accuracies.append((predicted.argmax(1) == labels).mean() * 100)`

However, it does continue training at the correct epoch when interrupting and running the command again.
(I am using Tensorflow 1.14 on a Titan X gpu.)

**Comment by david-berthelot** [comment]
When you say freezing, does it stay frozen? Because otherwise it's a normal behavior to pause while evaluation is being run.

**Comment by DewaldHoman** [comment]
Yes, it stays frozen indefinitely. But it seems to only hang after a few epochs on the first `valid` batch evaluation that is passed to `classify_op` during the tf session.

**Comment by david-berthelot** [comment]
I'm not sure how to go about this given I cannot reproduce it. Did you find anything new?

**Comment by david-berthelot** [comment]
May be this is a problem specific to your machine. Closing since I cannot reproduce.
